packages/jest-fake-timers/src/legacyFakeTimers.ts

```diff
@@ -526,9 +526,8 @@ export default class FakeTimers<TimerRef> {
 
     switch (timer.type) {
       case 'timeout':
-        const callback = timer.callback;
```

Comment:
this seems like some sort of trick to make GC work correctly. I doubt it matters though, so I'm fine with the change 👍